transport/src/test/java/io/netty/channel/DefaultChannelPipelineTest.java

```diff
@@ -2283,7 +2283,7 @@ public EventExecutorGroup parent() {
 
         @Override
         public boolean inEventLoop(Thread thread) {
-            return false;
+            return true;
```

Comment:
Why this change ?

Reply:
With the flatter stack, the task we punt onto the event loop will itself check if it is on the event loop, and punt if not. Hence, we need the executor to give `true` here at one point.

Reply:
Added a proper implementation.